Team417_2020/src/main/java/org/firstinspires/ftc/team417_2020/AutoCompetition.java

```diff
@@ -0,0 +1,131 @@
+package org.firstinspires.ftc.team417_2020;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.firstinspires.ftc.team417_2020.Resources.Toggler;
+import org.openftc.easyopencv.OpenCvCamera;
+import org.openftc.easyopencv.OpenCvCameraFactory;
+import org.openftc.easyopencv.OpenCvCameraRotation;
+
+@Autonomous(name="Competition Auto")
+public class AutoCompetition extends MasterAutonomous{
+
+    int allianceSide = 1;
+    Toggler rightBumper = new Toggler();
+
+    RingDetectionOpenCV ringDetector = new RingDetectionOpenCV();
+    OpenCvCamera webcam;
+    int numRings;
+
+
+
+    @Override
+    public void runOpMode() throws InterruptedException {
+        autoInitializeRobot();
+        initializeRingCounter();
+
+
+        telemetry.addLine("waiting for start");
+        telemetry.update();
+
+        while (!opModeIsActive()) {
+            countRings();
+            pickAllianceSide();
+            telemetry.update();
+        }
+
+
+        waitForStart();
+
+        // 0 rings -> Zone A
+        // 1 rings -> Zone B
+        // 4 rings -> Zone C
+        if (numRings == 0) {
```

Comment:
You don't have to change this, but sometimes it can look a little cleaner to use a switch statement when you're checking a value against a bunch of numbers.